adapters/repos/db/aggregator/iterator.go

```diff
@@ -0,0 +1,247 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package aggregator
+
+import (
+	"bytes"
+	"runtime"
+
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/sroar"
+	"github.com/weaviate/weaviate/adapters/repos/db/lsmkv"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+)
+
+func iteratorConcurrentlyReplace(b *lsmkv.Bucket, aggregateFunc func(b []byte) error, logger logrus.FieldLogger) error {
+	// we're looking at the whole object, so this is neither a Set, nor a Map, but
+	// a Replace strategy
+
+	seedCount := 2*runtime.GOMAXPROCS(0) - 1
```

Comment:
just curious: why `-1`?

Reply:
I copied that from `adapters/repos/db/vector/flat/compressed_parallel_iterator.go` don't think the -1 has any meaning. I think I will remove it to avoid confusion